### PR TITLE
lcm and gcd funtions for numpy front-end

### PR DESCRIPTION
**Pull request by hemanthkumarmudedla**
lcm #4151 
gcd #4155

### CODE REVIEW

**Comment by hemanthkumarmudedla** [comment]
Need some help.

**Comment by hirwa-nshuti** [comment]
Hi @hemanthkumarmudedla 
Here are some comments to take into consideration.
* First you didn't need to create extra files for the functions you were implementing as we have done that for you you can choose the proper file to implement a function into based on the type of the function.
*  Also do note a few of the following comments.

File: ivy/functional/frontends/numpy/mathematical_functions/lcm_and_gcd.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    where=True,
+    casting="same_kind",
+    order="k",
+    dtype=None,
+    subok=True,
+):
+    if dtype:
+        x1 = ivy.astype(ivy.array(x1), ivy.as_ivy_dtype(dtype))
+        x2 = ivy.astype(ivy.array(x2), ivy.as_ivy_dtype(dtype))
+    ret = ivy.lcm(x1, x2, out=out)
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/mathematical_functions/lcm_and_gcd.py:19]: We don't have this function in ivy, which means that this function will need to be implemented compositionally by referring to the contribution guide here: [Frontend guide](https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends.html)

File: ivy/functional/frontends/numpy/mathematical_functions/lcm_and_gcd.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    where=True,
+    casting="same_kind",
+    order="k",
+    dtype=None,
+    subok=True,
+):
+    if dtype:
+        x1 = ivy.astype(ivy.array(x1), ivy.as_ivy_dtype(dtype))
+        x2 = ivy.astype(ivy.array(x2), ivy.as_ivy_dtype(dtype))
+    ret = ivy.gcd(x1, x2, out=out)
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/mathematical_functions/lcm_and_gcd.py:40]: The same goes for this function too it needs to be implemented compositionally.

File: ivy/functional/frontends/numpy/mathematical_functions/__init__.py [hunk 3]
```diff
@@ -23,3 +23,5 @@
 from .extrema_finding import *
 from . import miscellaneous
 from .miscellaneous import *
+from . import lcm_and_gcd
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/mathematical_functions/__init__.py:26]: Not needed you will need to follow the current skeleton files structure.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
And for a better understanding of how frontend tests are implemented, head to this documentation guide on frontend tests [Ivy Frontend Tests](https://lets-unify.ai/ivy/deep_dive/17_ivy_frontends_tests.html)

**Pull request closed without merging**